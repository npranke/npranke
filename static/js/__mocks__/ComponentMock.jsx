function ComponentMock(props) {
    return (
        <div className="component-mock" role="document">
            {
                `${Object.keys(props).toString()} `
                + `${Object.values(props).toString()}`
            }
        </div>
    )
}

export default ComponentMock
