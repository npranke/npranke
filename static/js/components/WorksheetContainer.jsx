import { NavLink } from 'react-router-dom'
import PropTypes from 'prop-types'
import React from 'react'

import Code from '../../img/icon-code.png'
import { getLocationPageTitle, sendEvent } from '../utils'
import Info from '../../img/icon-info.png'

class WorksheetContainer extends React.Component {
    constructor(props) {
        super(props)

        this.state = { visible: 'worksheet' }

        this.infoPanel = React.createRef()
        this.worksheetPanel = React.createRef()
        this.gistPanel = React.createRef()

        this.infoTab = React.createRef()
        this.worksheetTab = React.createRef()
        this.gistTab = React.createRef()

        this.headerClickHandler = this.headerClickHandler.bind(this)
        this.headerKeyUpHandler = this.headerKeyUpHandler.bind(this)
        this.updateVisible = this.updateVisible.bind(this)
    }

    componentDidMount() {
        document.title = getLocationPageTitle(this.props.worksheet.title)

        this.updateVisible(this.props.location.hash)
    }

    componentDidUpdate(prevProps) {
        if (this.props.location.hash !== prevProps.location.hash) {
            this.updateVisible(this.props.location.hash)
        }
    }

    updateVisible(locationHash) {
        if (locationHash === '#info') {
            this.setState({ visible: 'info' })
        } else if (locationHash === '#gist') {
            this.setState({ visible: 'gist' })
        } else {
            this.setState({ visible: 'worksheet' })
        }
    }

    headerClickHandler(event) {
        const { section } = event.currentTarget.dataset

        sendEvent(`${this.props.worksheet.title}`, 'navigate', section)

        this.setState({ visible: section })
    }

    headerKeyUpHandler(event) {
        const { section } = event.currentTarget.dataset

        if (event.key === 'Enter' || event.key === ' ') {
            this.setState(
                { visible: section },
                () => {
                    if (section === 'info') {
                        this.infoPanel.current.focus()
                    } else if (section === 'gist') {
                        this.gistPanel.current.focus()
                    } else {
                        this.worksheetPanel.current.focus()
                    }
                },
            )
        } else if (event.key === 'ArrowRight') {
            if (section === 'info') {
                this.worksheetTab.current.focus()
            } else if (section === 'worksheet') {
                this.gistTab.current.focus()
            }
        } else if (event.key === 'ArrowLeft') {
            if (section === 'gist') {
                this.worksheetTab.current.focus()
            } else if (section === 'worksheet') {
                this.infoTab.current.focus()
            }
        }
    }

    render() {
        const infoSection = (
            <div
                className="worksheet-section info-container"
                id="info-tabpanel"
                role="tabpanel"
                aria-expanded={ this.state.visible === 'info' }
                tabIndex="0"
                ref={ this.infoPanel }
            >
                <div className="worksheet-info">
                    <div className="worksheet-info-inner">
                        { this.props.worksheet.infoComponent }
                    </div>
                </div>
            </div>
        )

        const worksheetSection = (
            <div
                className={
                    'worksheet-section '
                    + `${this.props.worksheet.pathTitle}-container`
                }
                id="worksheet-tabpanel"
                role="tabpanel"
                aria-expanded={ this.state.visible === 'worksheet' }
                tabIndex="0"
                ref={ this.worksheetPanel }
            >
                { this.props.worksheet.component }
            </div>
        )

        const gistSection = (
            <div
                className="worksheet-section gist-container"
                id="gist-tabpanel"
                role="tabpanel"
                aria-expanded={ this.state.visible === 'gist' }
                tabIndex="0"
                ref={ this.gistPanel }
            >
                <iframe
                    id="worksheet-gist"
                    title={ `${this.props.worksheet.properTitle} gist` }
                    sandbox="allow-scripts allow-top-navigation"
                    referrerPolicy="no-referrer"
                    src={ `/gists/${this.props.worksheet.pathTitle}` }
                >
                </iframe>
            </div>
        )

        const sections = {
            info: infoSection,
            gist: gistSection,
            worksheet: worksheetSection,
        }

        return (
            <main className="workbook worksheet-container">
                <div className="worksheet-header" role="tablist">
                    <span
                        className="section-button"
                        role="tab"
                        id="info-tab"
                        aria-controls="info-tabpanel"
                        aria-selected={ this.state.visible === 'info' }
                    >
                        <NavLink
                            to="#info"
                            target="_self"
                            rel="noreferrer"
                            isActive={ (match, location) => {
                                return location.hash === '#info'
                            } }
                            className="local-navlink local-navlink-info"
                            id="info-tab-navlink"
                            onClick={ this.headerClickHandler }
                            onKeyUp={ this.headerKeyUpHandler }
                            data-section="info"
                            aria-current="true"
                            innerRef={ this.infoTab }
                        >
                            <img
                                src={ Info }
                                className="icon icon-info"
                                alt="Info icon"
                            />
                            <span className="section-button-text">
                                info
                            </span>
                        </NavLink>
                    </span>
                    <span
                        className="section-button"
                        role="tab"
                        id="worksheet-tab"
                        aria-controls="worksheet-tabpanel"
                        aria-selected={ this.state.visible === 'worksheet' }
                    >
                        <NavLink
                            to="#"
                            target="_self"
                            rel="noreferrer"
                            isActive={ (match, location) => {
                                return location.hash === ''
                            } }
                            className="local-navlink local-navlink-worksheet"
                            id="worksheet-tab-navlink"
                            onClick={ this.headerClickHandler }
                            onKeyUp={ this.headerKeyUpHandler }
                            data-section="worksheet"
                            aria-current="true"
                            innerRef={ this.worksheetTab }
                        >
                            <img
                                src={ this.props.worksheet.icon }
                                className={
                                    'icon '
                                    + `icon-${this.props.worksheet.pathTitle}`
                                }
                                alt={
                                    `${this.props.worksheet.properTitle} icon`
                                }
                            />
                            <span className="section-button-text">
                                { this.props.worksheet.title }
                            </span>
                        </NavLink>
                    </span>
                    <span
                        className="section-button"
                        role="tab"
                        id="gist-tab"
                        aria-controls="gist-tabpanel"
                        aria-selected={ this.state.visible === 'gist' }
                    >
                        <NavLink
                            to="#gist"
                            target="_self"
                            rel="noreferrer"
                            isActive={ (match, location) => {
                                return location.hash === '#gist'
                            } }
                            className="local-navlink local-navlink-gist"
                            id="gist-tab-navlink"
                            onClick={ this.headerClickHandler }
                            onKeyUp={ this.headerKeyUpHandler }
                            data-section="gist"
                            aria-current="true"
                            innerRef={ this.gistTab }
                        >
                            <img
                                src={ Code }
                                className="icon icon-code"
                                alt="Code icon"
                            />
                            <span className="section-button-text">
                                gist
                            </span>
                        </NavLink>
                    </span>
                </div>
                { sections[this.state.visible] }
            </main>
        )
    }
}

WorksheetContainer.propTypes = {
    location: PropTypes.shape({
        hash: PropTypes.string,
    }).isRequired,
    worksheet: PropTypes.shape({
        component: PropTypes.element.isRequired,
        icon: PropTypes.node.isRequired,
        infoComponent: PropTypes.element.isRequired,
        pathTitle: PropTypes.string.isRequired,
        properTitle: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
    }).isRequired,
}

export default WorksheetContainer
