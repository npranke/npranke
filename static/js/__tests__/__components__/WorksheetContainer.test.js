import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { mount, shallow } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import React from 'react'
import { render } from '@testing-library/react'

import ComponentMock from '@__mocks__/ComponentMock'
import ImageMock from '@__mocks__/ImageMock'

import WorksheetContainer from '@components/WorksheetContainer'

import * as utils from '@utils'

Enzyme.configure({ adapter: new Adapter() })

const mockLocation = { hash: '' }

jest.mock('react-router-dom-v5-compat', () => {
    return {
        ...jest.requireActual('react-router-dom-v5-compat'),
        useLocation: jest.fn(() => { return mockLocation }),
    }
})

const worksheet = {
    component: <ComponentMock />,
    icon: ImageMock,
    infoComponent: <ComponentMock />,
    pathTitle: 'component',
    properTitle: 'Component',
    title: 'component',
}

describe('WorksheetContainer', () => {
    beforeEach(() => {
        mockLocation.hash = ''
    })

    test('sets document title', () => {
        mount(
            <WorksheetContainer worksheet={ worksheet } />,
            { wrappingComponent: MemoryRouter },
        )

        expect(document.title).toContain('component')
    })

    test('has three tabs', () => {
        const worksheetContainer = shallow(
            <WorksheetContainer worksheet={ worksheet } />,
        )

        expect(
            worksheetContainer.find('.section-button'),
        ).toHaveLength(3)

        worksheetContainer.find('.section-button').forEach(
            (sectionButton) => {
                expect(sectionButton.props().role).toEqual('tab')
            },
        )
    })

    test('has one tabpanel', () => {
        const worksheetContainer = shallow(
            <WorksheetContainer worksheet={ worksheet } />,
        )

        expect(
            worksheetContainer.find('.worksheet-section'),
        ).toHaveLength(1)

        expect(
            worksheetContainer.find('.worksheet-section').props().role,
        ).toEqual('tabpanel')
    })

    describe('sends pageview', () => {
        beforeAll(() => {
            utils.sendPageview = jest.fn()
        })

        afterEach(() => {
            utils.sendPageview.mockReset()
        })

        test('sends pageview when component mounts', () => {
            mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { wrappingComponent: MemoryRouter },
            )

            expect(utils.sendPageview).toHaveBeenCalledTimes(1)
        })

        test('sends pageview when location hash changes', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { wrappingComponent: MemoryRouter },
            )

            expect(utils.sendPageview).toHaveBeenCalledTimes(1)

            mockLocation.hash = '#gist'
            worksheetContainer.setProps({})

            expect(utils.sendPageview).toHaveBeenCalledTimes(2)
        })
    })

    describe('headerClickHandler', () => {
        test('updates aria-selected on with click on info', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#info-tab NavLink').simulate('click')

            expect(
                worksheetContainer.find('#info-tab').props()['aria-selected'],
            ).toBe(true)
            expect(
                worksheetContainer.find(
                    '#worksheet-tab',
                ).props()['aria-selected'],
            ).toBe(false)
            expect(
                worksheetContainer.find('#gist-tab').props()['aria-selected'],
            ).toBe(false)
        })

        test('updates aria-selected on with click on gist', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#gist-tab NavLink').simulate('click')

            expect(
                worksheetContainer.find('#info-tab').props()['aria-selected'],
            ).toBe(false)
            expect(
                worksheetContainer.find(
                    '#worksheet-tab',
                ).props()['aria-selected'],
            ).toBe(false)
            expect(
                worksheetContainer.find('#gist-tab').props()['aria-selected'],
            ).toBe(true)
        })
    })

    describe('headerKeyUpHandler', () => {
        beforeEach(() => {
            document.body.innerHTML = '<div id="fastener"></div>'
        })

        test('enter key up event on info section button', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { attachTo: fastener, wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#info-tab NavLink').simulate(
                'keyup',
                { key: 'Enter' },
            )

            expect(
                document.activeElement.id,
            ).toEqual('info-tabpanel')
        })

        test('enter key up event on worksheet section button', () => {
            mockLocation.hash = '#info'

            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { attachTo: fastener, wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#worksheet-tab NavLink').simulate(
                'keyup',
                { key: 'Enter' },
            )

            expect(
                document.activeElement.id,
            ).toEqual('worksheet-tabpanel')
        })

        test('enter key up event on gist section button', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { attachTo: fastener, wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#gist-tab NavLink').simulate(
                'keyup',
                { key: 'Enter' },
            )

            expect(
                document.activeElement.id,
            ).toEqual('gist-tabpanel')
        })

        test('spacebar key up event on info section button', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { attachTo: fastener, wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#info-tab NavLink').simulate(
                'keyup',
                { key: ' ' },
            )

            expect(
                document.activeElement.id,
            ).toEqual('info-tabpanel')
        })

        test('spacebar key up event on worksheet section button', () => {
            mockLocation.hash = '#info'

            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { attachTo: fastener, wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#worksheet-tab NavLink').simulate(
                'keyup',
                { key: ' ' },
            )

            expect(
                document.activeElement.id,
            ).toEqual('worksheet-tabpanel')
        })

        test('spacebar key up event on gist section button', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { attachTo: fastener, wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#gist-tab NavLink').simulate(
                'keyup',
                { key: ' ' },
            )

            expect(
                document.activeElement.id,
            ).toEqual('gist-tabpanel')
        })

        test('arrow right key up event on info section button', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { attachTo: fastener, wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#info-tab NavLink').simulate(
                'keyup',
                { key: 'ArrowRight' },
            )

            expect(
                document.activeElement.id,
            ).toEqual('worksheet-tab-navlink')
        })

        test('arrow right key up event on worksheet section button', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { attachTo: fastener, wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#worksheet-tab NavLink').simulate(
                'keyup',
                { key: 'ArrowRight' },
            )

            expect(
                document.activeElement.id,
            ).toEqual('gist-tab-navlink')
        })

        test('arrow left key up event on worksheet section button', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { attachTo: fastener, wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#worksheet-tab NavLink').simulate(
                'keyup',
                { key: 'ArrowLeft' },
            )

            expect(
                document.activeElement.id,
            ).toEqual('info-tab-navlink')
        })

        test('arrow left key up event on gist section button', () => {
            const worksheetContainer = mount(
                <WorksheetContainer worksheet={ worksheet } />,
                { attachTo: fastener, wrappingComponent: MemoryRouter },
            )

            worksheetContainer.find('#gist-tab NavLink').simulate(
                'keyup',
                { key: 'ArrowLeft' },
            )

            expect(
                document.activeElement.id,
            ).toEqual('worksheet-tab-navlink')
        })
    })
})

describe('WorksheetContainer snapshot', () => {
    beforeEach(() => {
        mockLocation.hash = ''
    })

    test('matches snapshot when location hash is empty', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <WorksheetContainer worksheet={ worksheet } />
            </MemoryRouter>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('matches snapshot when location hash is #info', () => {
        mockLocation.hash = '#info'

        const { asFragment } = render(
            <MemoryRouter>
                <WorksheetContainer worksheet={ worksheet } />
            </MemoryRouter>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('matches snapshot when location hash is #gist', () => {
        mockLocation.hash = '#gist'

        const { asFragment } = render(
            <MemoryRouter>
                <WorksheetContainer worksheet={ worksheet } />
            </MemoryRouter>,
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
