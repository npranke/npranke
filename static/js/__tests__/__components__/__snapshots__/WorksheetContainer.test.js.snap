// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorksheetContainer snapshot matches snapshot when location hash is #gist 1`] = `
<main
  className="workbook worksheet-container"
>
  <div
    className="worksheet-header"
    role="tablist"
  >
    <span
      aria-controls="info-tabpanel"
      aria-selected={false}
      className="section-button"
      id="info-tab"
      role="tab"
    >
      <NavLink
        aria-current="true"
        className="local-navlink local-navlink-info"
        data-section="info"
        id="info-tab-navlink"
        innerRef={
          Object {
            "current": null,
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        onKeyUp={[Function]}
        rel="noreferrer"
        target="_self"
        to="#info"
      >
        <img
          alt="Info icon"
          className="icon icon-info"
          src="image-mock"
        />
        <span
          className="section-button-text"
        >
          info
        </span>
      </NavLink>
    </span>
    <span
      aria-controls="worksheet-tabpanel"
      aria-selected={false}
      className="section-button"
      id="worksheet-tab"
      role="tab"
    >
      <NavLink
        aria-current="true"
        className="local-navlink local-navlink-worksheet"
        data-section="worksheet"
        id="worksheet-tab-navlink"
        innerRef={
          Object {
            "current": null,
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        onKeyUp={[Function]}
        rel="noreferrer"
        target="_self"
        to="#"
      >
        <img
          alt="Component icon"
          className="icon icon-component"
          src="image-mock"
        />
        <span
          className="section-button-text"
        >
          component
        </span>
      </NavLink>
    </span>
    <span
      aria-controls="gist-tabpanel"
      aria-selected={true}
      className="section-button"
      id="gist-tab"
      role="tab"
    >
      <NavLink
        aria-current="true"
        className="local-navlink local-navlink-gist"
        data-section="gist"
        id="gist-tab-navlink"
        innerRef={
          Object {
            "current": null,
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        onKeyUp={[Function]}
        rel="noreferrer"
        target="_self"
        to="#gist"
      >
        <img
          alt="Code icon"
          className="icon icon-code"
          src="image-mock"
        />
        <span
          className="section-button-text"
        >
          gist
        </span>
      </NavLink>
    </span>
  </div>
  <div
    aria-expanded={true}
    className="worksheet-section gist-container"
    id="gist-tabpanel"
    role="tabpanel"
    tabIndex="0"
  >
    <iframe
      id="worksheet-gist"
      referrerPolicy="no-referrer"
      sandbox="allow-scripts allow-top-navigation"
      src="/gists/component"
      title="Component gist"
    />
  </div>
</main>
`;

exports[`WorksheetContainer snapshot matches snapshot when location hash is #info 1`] = `
<main
  className="workbook worksheet-container"
>
  <div
    className="worksheet-header"
    role="tablist"
  >
    <span
      aria-controls="info-tabpanel"
      aria-selected={true}
      className="section-button"
      id="info-tab"
      role="tab"
    >
      <NavLink
        aria-current="true"
        className="local-navlink local-navlink-info"
        data-section="info"
        id="info-tab-navlink"
        innerRef={
          Object {
            "current": null,
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        onKeyUp={[Function]}
        rel="noreferrer"
        target="_self"
        to="#info"
      >
        <img
          alt="Info icon"
          className="icon icon-info"
          src="image-mock"
        />
        <span
          className="section-button-text"
        >
          info
        </span>
      </NavLink>
    </span>
    <span
      aria-controls="worksheet-tabpanel"
      aria-selected={false}
      className="section-button"
      id="worksheet-tab"
      role="tab"
    >
      <NavLink
        aria-current="true"
        className="local-navlink local-navlink-worksheet"
        data-section="worksheet"
        id="worksheet-tab-navlink"
        innerRef={
          Object {
            "current": null,
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        onKeyUp={[Function]}
        rel="noreferrer"
        target="_self"
        to="#"
      >
        <img
          alt="Component icon"
          className="icon icon-component"
          src="image-mock"
        />
        <span
          className="section-button-text"
        >
          component
        </span>
      </NavLink>
    </span>
    <span
      aria-controls="gist-tabpanel"
      aria-selected={false}
      className="section-button"
      id="gist-tab"
      role="tab"
    >
      <NavLink
        aria-current="true"
        className="local-navlink local-navlink-gist"
        data-section="gist"
        id="gist-tab-navlink"
        innerRef={
          Object {
            "current": null,
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        onKeyUp={[Function]}
        rel="noreferrer"
        target="_self"
        to="#gist"
      >
        <img
          alt="Code icon"
          className="icon icon-code"
          src="image-mock"
        />
        <span
          className="section-button-text"
        >
          gist
        </span>
      </NavLink>
    </span>
  </div>
  <div
    aria-expanded={true}
    className="worksheet-section info-container"
    id="info-tabpanel"
    role="tabpanel"
    tabIndex="0"
  >
    <div
      className="worksheet-info"
    >
      <div
        className="worksheet-info-inner"
      >
        <ComponentMock />
      </div>
    </div>
  </div>
</main>
`;

exports[`WorksheetContainer snapshot matches snapshot when location hash is empty 1`] = `
<main
  className="workbook worksheet-container"
>
  <div
    className="worksheet-header"
    role="tablist"
  >
    <span
      aria-controls="info-tabpanel"
      aria-selected={false}
      className="section-button"
      id="info-tab"
      role="tab"
    >
      <NavLink
        aria-current="true"
        className="local-navlink local-navlink-info"
        data-section="info"
        id="info-tab-navlink"
        innerRef={
          Object {
            "current": null,
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        onKeyUp={[Function]}
        rel="noreferrer"
        target="_self"
        to="#info"
      >
        <img
          alt="Info icon"
          className="icon icon-info"
          src="image-mock"
        />
        <span
          className="section-button-text"
        >
          info
        </span>
      </NavLink>
    </span>
    <span
      aria-controls="worksheet-tabpanel"
      aria-selected={true}
      className="section-button"
      id="worksheet-tab"
      role="tab"
    >
      <NavLink
        aria-current="true"
        className="local-navlink local-navlink-worksheet"
        data-section="worksheet"
        id="worksheet-tab-navlink"
        innerRef={
          Object {
            "current": null,
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        onKeyUp={[Function]}
        rel="noreferrer"
        target="_self"
        to="#"
      >
        <img
          alt="Component icon"
          className="icon icon-component"
          src="image-mock"
        />
        <span
          className="section-button-text"
        >
          component
        </span>
      </NavLink>
    </span>
    <span
      aria-controls="gist-tabpanel"
      aria-selected={false}
      className="section-button"
      id="gist-tab"
      role="tab"
    >
      <NavLink
        aria-current="true"
        className="local-navlink local-navlink-gist"
        data-section="gist"
        id="gist-tab-navlink"
        innerRef={
          Object {
            "current": null,
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        onKeyUp={[Function]}
        rel="noreferrer"
        target="_self"
        to="#gist"
      >
        <img
          alt="Code icon"
          className="icon icon-code"
          src="image-mock"
        />
        <span
          className="section-button-text"
        >
          gist
        </span>
      </NavLink>
    </span>
  </div>
  <div
    aria-expanded={true}
    className="worksheet-section component-container"
    id="worksheet-tabpanel"
    role="tabpanel"
    tabIndex="0"
  >
    <ComponentMock />
  </div>
</main>
`;
